File: two_pointers/two_sum_2/two_sum_sorted.py
class Solution:
    def twoSum(self, numbers, target):
        start = 0
        end = len(numbers) - 1
        while start < end:
            s = numbers[start] + numbers[end]
            if s == target:
                return [start+1, end+1]
            if s < target:
                start+=1
            else:
                end-=1

File: two_pointers/two_sum_2/two_sum_sorted.c
#include <stdlib.h>
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* twoSum(int* numbers, int numbersSize, int target, int* returnSize){
    *returnSize = 2;
    int start = 0;
    int end = numbersSize -1;
    int *res = malloc(2 * sizeof(int));
    while (start < end) {
        int curr = numbers[start] + numbers[end];
        if (curr == target) {
            res[0] = start + 1;
            res[1] = end + 1;
            return res;
        }

        if (curr > target) {
            end--;
        } else {
            start++;
        }
    }
    return res;
}